dagger-core/src/main/java/com/gotocompany/dagger/core/sink/SinkOrchestrator.java

```diff
@@ -107,11 +107,19 @@ protected Properties getProducerProperties(Configuration configuration) {
             kafkaProducerConfigs.setProperty(Constants.SINK_KAFKA_MAX_REQUEST_SIZE_KEY, Constants.SINK_KAFKA_MAX_REQUEST_SIZE_DEFAULT);
         }
         String lingerMs = configuration.getString(Constants.SINK_KAFKA_LINGER_MS_KEY, Constants.SINK_KAFKA_LINGER_MS_DEFAULT);
+        validateLingerMs(lingerMs);
         kafkaProducerConfigs.setProperty(Constants.SINK_KAFKA_LINGER_MS_CONFIG_KEY, lingerMs);
 
         return kafkaProducerConfigs;
     }
 
+    private void validateLingerMs(String lingerMs) {
+        int x = Integer.parseInt(lingerMs);
+        if (x < 0) {
+            throw new IllegalArgumentException("Linger Ms value should be an unsigned integer");
```

Comment:
actually the range is long, not int, and also i meant to check the type like its not a float or string. You can just put 
`        try {
            Long.parseLong(lingerMs);
        } catch (Exception e) {
            throw new IllegalArgumentException("Provided value for Linger Ms : " + lingerMs+ " is not a valid long");
        }`

Reply:
Quite strange, the data type in documentation is long but internally it converts the long into an integer. [Reference](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L550).
Should we keep it int in such a case ? Since, anyways we are abstracting the config value away from what it actually is provided in the kafka producer and int is anyways a subset of long. WDYT @sumitaich1998

Reply:
yeah ok int should be fine , so you can just do -
`        try { Integer.parseInt(lingerMs); } catch (Exception e) { throw new IllegalArgumentException("Provided value for Linger Ms : " + lingerMs+ " is not a valid integer"); }`

Reply:
The error that parseInt throws is NumberFormatException which is a subclass of IllegalArgumentException. Any reason to have a custom message. Since, the default message would be more specific as to what was the mismatch.

Reply:
so that the user can debug easily , you can add the NumberFormatException's message as well like - 
`        try { Integer.parseInt(lingerMs); } catch (Exception e) { throw new IllegalArgumentException("Provided value for Linger Ms : " + lingerMs+ " is not a valid integer , Error: " + e.getMessage() ); }`

Reply:
I have made the changes.